squarelet/organizations/views.py

```diff
@@ -173,31 +233,90 @@ class ManageMembers(OrganizationAdminMixin, UpdateView):
+        elif action == 'removeuser':
+            return self._handle_user(request, lambda membership: membership.delete(), 'Removed user')
+
+    def _handle_add_member(self, request, addmember_form):
+        if not addmember_form.is_valid():
+            # Ensure the email is valid
+            messages.error(request, "Please enter a valid email address")
+        else:
+            # Ensure the org has capacity
+            organization = self.get_object()
+            if organization.user_count() >= organization.max_users:
+                messages.error(
+                    request, "You need to increase your max users to invite another member"
+                )
+            else:
+                # Create an invitation and send it to the given email address
+                invitation = Invitation.objects.create(
+                    organization=organization, email=addmember_form.cleaned_data["email"]
+                )
+                invitation.send()
+                messages.success(self.request, "Invitation sent")
+        return redirect('organizations:manage-members', slug=self.get_object().slug)
+
+    def _handle_invite(self, request, inviteFn, successMessage):
+        try:
+            inviteid = request.POST.get('inviteid')
+            assert(inviteid is not None)
+            invite = Invitation.objects.get(pk=inviteid)
+            assert(invite.organization == self.get_object())
+            inviteFn(invite)
+            messages.success(self.request, successMessage)
+        except:
+            return self._bad_call(request)
+        return redirect('organizations:manage-members', slug=self.get_object().slug)
+
+    def _handle_user(self, request, membershipFn, successMessage):
+        try:
+            userid = request.POST.get('userid')
+            assert(userid is not None)
+            membership = self.get_object().memberships.get(user_id=userid)
+            membershipFn(membership)
+            messages.success(self.request, successMessage)
+        except:
+            return self._bad_call(request)
+        return redirect('organizations:manage-members', slug=self.get_object().slug)
+
+    def _bad_call(self, request):
+        messages.error(self.request, 'An unexpected error occurred')
+        return redirect('organizations:manage-members', slug=self.get_object().slug)
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         form = context["form"]
-        context["users"] = [
-            (u, form[f"admin-{u.pk}"], form[f"remove-{u.pk}"])
-            for u in self.object.users.all()
-        ]
+        context["admin"] = self.request.user
+        users = self.object.users.all().order_by('created_at')
+        context["users"] = [(u, self.get_object().memberships.get(user_id=u.id).admin) for u in users]
+        context["requested_invitations"] = self.object.invitations.get_requested().order_by('created_at')
```

Comment:
Probably makes sense to just make `created_at` the default ordering for invitations